src/test/java/jimmy/mcgymmy/model/food/FatTest.java

```diff
@@ -0,0 +1,44 @@
+package jimmy.mcgymmy.model.food;
+
+import static jimmy.mcgymmy.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class FatTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NumberFormatException.class, () -> new Fat(null));
+    }
+
+    @Test
+    public void constructor_invalidEmail_throwsIllegalArgumentException() {
+        String invalidFat = "12@123";
+        assertThrows(NumberFormatException.class, () -> new Fat(Integer.parseInt(invalidFat)));
+    }
+
+    @Test
+    public void isValid() {
+        // null fat
+        assertThrows(NullPointerException.class, () -> Fat.isValid(null));
+
+        // blank fat
+        Assertions.assertFalse(Fat.isValid("")); // empty string
+        Assertions.assertFalse(Fat.isValid(" ")); // spaces only
+
+        // invalid parts
+        Assertions.assertFalse(Fat.isValid("peterjack@-")); // invalid fat name
+        Assertions.assertFalse(Fat.isValid("peterjack@exam_ple.com")); // underscore in fat name
```

Comment:
should change all these to resemble the typical values, except with an underscore, space etc.

i.e. each test should only have 1 reason to fail.